draft-ietf-ice-pac.xml

```diff
@@ -172,37 +179,41 @@
+      </t>
+      <t>
+        A full ICE implementation starts timer Tpac once it has received the set of remote candidates from
+        the remote peer and has sent back its set of local candidates, as described in Section 5 of <xref target="RFC8445" 
+        pageno="false" format="default"/>.
       </t>
       <t>
+        The timeout value of timer Tpac SHOULD be set to the timeout value of an ICE connectivity
+        check transaction associated with the ICE session.
         This timeout value is chosen to roughly coincide with the maximum
         possible duration of ICE connectivity checks from the remote peer,
         which, if successful, could create peer-reflexive candidates. Because
-        the ICE agent doesn't know the exact number of candidate pairs and pacing
-        interval in use by the remote side, this timeout value is simply a
-        guess, albeit an educated one. Regardless, for this particular problem,
-        the desired benefits will be realized as long as the agent waits
-        some reasonable amount of time.
+        the ICE agent does not know the exact number of candidate pairs and pacing
+        interval in use by the peer, this timeout value is simply a
+        guess, albeit an educated one.
+      </t>
+      <t>
+        While timer Tpac is running, the ICE agent MUST NOT set the state of a
+        checklist to Failed, even if there are no more candidate pairs to check
+        in the checklist. As a result, the ICE agent will not remove data streams,
+        or set the state of the ICE session to Failed, while timer Tpac is still running.
       </t>
       <t>
-        While the timer is running, if a checklist has no pairs left to check,
-        i.e., there are no pairs that are not in the failed state, the ICE
-        agent MUST not conclude that ICE processing has failed, and MUST wait
-        for the timer to elapse before doing so.
+        While timer Tpac is running, if the ICE agent has set the state of each checklist
+        to Completed, the ICE agent can terminate the timer and set the state of the ICE session
+        to Completed, following normal ICE procedures.
       </t>
       <t>
-        One consequence of this behavior is that in cases where ICE should fail,
-        e.g., where both sides provide candidates with unresolvable FQDNs
-        ICE will no longer fail immediately, and only fail when the
-        aforementioned timer expires. However, because most ICE scenarios
-        require an extended period of time to determine failure, the
-        fact that some specific scenarios no longer fail fast should have
-        minimal application impact, if any.
+        If timer Tpac times out while the state of at least one checklist is
+        Running, the ICE agent MUST continue the ICE processing, and will later 
+        conclude the ICE processing and set the state of the ICE session to either Completed
+        or Failed, follwoing normal ICE procedurs.
```

Comment:
It's not clear to me why this text needs to change

Reply:
1. It clarifies that the checklist state is not set to Failed while the timer is still running, as peer-reflexive candidates may still arrive.

2. One sentence said:

"While the timer is running, if a checklist has no pairs left to check,
i.e., there are no pairs that are not in the failed state,"

I am not really sure what that means. Lots of "no" and "not".

3. It explicitly covers that case what happens if the timer expires while ICE is still processing.